### ISSUE TITLE
Callback with $ref is not working

### ISSUE THREAD

**Issue by jmini** [swagger-api/swagger-parser#879]
Consider this input file:

```yaml
openapi: 3.0.0
info:
  title: Callback with ref Example
  version: 1.0.0
paths:
  /register:
    post:
      summary: Subscribe to a webhook
      requestBody:
        required: true
        content:
          application/json:
            schema:
              type: object
              properties:
                callbackUrl: # Callback URL
                  type: string
                  format: uri
                  example: https://myserver.com/send/callback/here
              required:
                - callbackUrl
      responses:
        '200':
          description: subscription successfully created
          content:
            application/json:
              schema:
                type: object
                description: subscription information
                required:
                  - subscriptionId
                properties:
                  subscriptionId:
                    description: unique identifier
                    type: string
                    example: 2531329f-fb09-4ef7-887e-84e648214436
      callbacks:
        myEvent:
          $ref: '#/components/callbacks/callbackEvent'
components:
  callbacks:
    callbackEvent:
      '{$request.body#/callbackUrl}':
        post:
          requestBody: # Contents of the callback message
            required: true
            content:
              application/json:
                schema:
                  type: object
                  properties:
                    message:
                      type: string
                      example: Some event happened
                  required:
                    - message
          responses:
            '200':
              description: ok
```

Test code:

```java
final String inputSpec = /* path */;

OpenAPIParser openApiParser = new OpenAPIParser();
ParseOptions options = new ParseOptions();
OpenAPI openAPI = openApiParser.readLocation(inputSpec, null, options).getOpenAPI();

String ref = openAPI.getPaths()
                       .get("/register")
                       .getPost()
                       .getCallbacks()
                       .get("myEvent")
                       .get$ref();
System.out.println(ref);
```

`ref` should contain: `#/components/callbacks/callbackEvent`

current result is null.

Tested with:

```xml
<dependency>
  <groupId>io.swagger.parser.v3</groupId>
  <artifactId>swagger-parser</artifactId>
  <version>2.0.5</version>
</dependency>
```

**Comment by ymohdriz** [comment]
Hi @jmini,

Reason for this issue is https://github.com/swagger-api/swagger-core/issues/2638 which is fixed in swagger-core but not reflected in swagger-parser.

I have raised PR to fix this issue https://github.com/swagger-api/swagger-parser/pull/880

Thanks,
Mohammed

**Comment by gracekarina** [comment]
Fixed by #880
